CMakeLists.txt

```diff
@@ -68,11 +73,19 @@ endif()
 # Find OpenSSL
 ###############################################################################
 
+option(OpenSSL_REQUIRED "Enable or disable the requirement of OpenSSL" ON)
 find_package(OpenSSL)
-include_directories(${OPENSSL_INCLUDE_DIR})
-set(LIBS ${LIBS} ${OPENSSL_LIBRARIES})
-if(NOT OPENSSL_FOUND)
-    add_definitions("-DCONF_NO_TLS")
+if(OPENSSL_FOUND)
+    include_directories(${OPENSSL_INCLUDE_DIR})
+    set(LIBS ${LIBS} ${OPENSSL_LIBRARIES})
+else()
+    if(OpenSSL_REQUIRED)
+        message(FATAL_ERROR "OpenSSL NOT found: use `-DOpenSSL_REQUIRED=OFF` to build without SSL support")
+    else()
+        if(NOT OPENSSL_FOUND)
```

Comment:
Is it possible to remove the second if in all versions, or would it break something?